tests/pipelines/test_pipeline.py

```diff
@@ -33,7 +33,9 @@
     chdir,
 )  # pylint: enable=unused-import
 
-_STEP_NAMES = ["ingest", "split", "train", "transform", "evaluate"]
+# _STEP_NAMES must contain all step names that are expected to be executed when
+# `pipeline.run(step=None)` is called
+_STEP_NAMES = ["ingest", "split", "train", "transform", "evaluate", "register"]
```

Comment:
Existing test already covers testing `self._get_default_step()`: `test_pipelines_execution_directory_is_managed_as_expected()` calls `p.run()` and asserts outputs are found for all steps in `_STEP_NAMES`.